python/sparkdl/param/converters.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import six
+
+import tensorflow as tf
+
+from pyspark.ml.param import TypeConverters
+
+import sparkdl.graph.utils as tfx
+import sparkdl.utils.keras_model as kmutil
+
+__all__ = ['SparkDLTypeConverters']
+
+def _get_strict_tensor_name(_maybe_tnsr_name):
+    assert isinstance(_maybe_tnsr_name, six.string_types), \
+        "must provide a strict tensor name as input, but got {}".format(type(_maybe_tnsr_name))
+    assert tfx.as_tensor_name(_maybe_tnsr_name) == _maybe_tnsr_name, \
+        "input {} must be a valid tensor name".format(_maybe_tnsr_name)
+    return _maybe_tnsr_name
+
+def _try_convert_tf_tensor_mapping(value, is_key_tf_tensor=True):
+    if isinstance(value, dict):
+        strs_pair_seq = []
+        for k, v in value.items():
+            # Check if the non-tensor value is of string type
+            _non_tnsr_str_val = v if is_key_tf_tensor else k
+            if not isinstance(_non_tnsr_str_val, six.string_types):
+                err_msg = 'expect string type for {}, but got {}'
+                raise TypeError(err_msg.format(_non_tnsr_str_val, type(_non_tnsr_str_val)))
+
+            # Check if the tensor name is actually valid
+            try:
+                if is_key_tf_tensor:
+                    _pair = (_get_strict_tensor_name(k), v)
+                else:
+                    _pair = (k, _get_strict_tensor_name(v))
+            except Exception as exc:
+                err_msg = "Can NOT convert {} (type {}) to tf.Tensor name: {}"
+                _not_tf_op = k if is_key_tf_tensor else v
+                raise TypeError(err_msg.format(_not_tf_op, type(_not_tf_op), exc))
+
+            strs_pair_seq.append(_pair)
+
+        return sorted(strs_pair_seq)
+
+    if is_key_tf_tensor:
+        raise TypeError("Could not convert %s to tf.Tensor name to str mapping" % type(value))
+    else:
+        raise TypeError("Could not convert %s to str to tf.Tensor name mapping" % type(value))
+
+
+class SparkDLTypeConverters(object):
+    @staticmethod
+    def toTFGraph(value):
```

Comment:
put docs for each of these methods. What are the inputs? What are the outputs? It is impossible in python to know that unless you follow through the code, and you will be hard pressed to know what these functions do in 6 months.

Reply:
mark as change doc => will modify as the last step